src/modules/tasks.js

```diff
@@ -0,0 +1,105 @@
+const tasks = JSON.parse(localStorage.getItem('tasks')) || [];
+
+function saveTasks() {
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+}
+
+export function renderTasks() {
+  const taskContainer = document.querySelector('.task-container');
+  taskContainer.innerHTML = '';
+
+  tasks.sort((a, b) => a.index - b.index);
+
+  tasks.forEach((task) => {
+    // Create the task item HTML element
+    const taskItem = document.createElement('li');
+    taskItem.classList.add('task-item', 'flex-row');
+
+    // Add the completed class if the task has been completed
+    if (task.completed) {
+      taskItem.classList.add('task-completed');
+    }
+
+    // Create the task checkbox
+    const taskCheckbox = document.createElement('input');
+    taskCheckbox.type = 'checkbox';
+    taskCheckbox.checked = task.completed;
+    taskItem.appendChild(taskCheckbox);
+
+    // Create the task description element
+    const taskDescription = document.createElement('p');
+    taskDescription.innerText = task.description;
+    taskDescription.contentEditable = false;
+    taskDescription.addEventListener('input', () => {
+      task.description = taskDescription.innerText;
+      saveTasks();
+    });
+    taskItem.appendChild(taskDescription);
+
+    // Create the task menu icon and trash icon
+    const taskMenu = document.createElement('i');
+    taskMenu.classList.add('las', 'la-ellipsis-v', 'btn');
+    const trashIcon = document.createElement('i');
+    trashIcon.classList.add('las', 'la-trash-alt', 'btn');
+    trashIcon.classList.add('hidden');
+
+    // Add event listener to handle task editing functionality
+    taskMenu.addEventListener('click', () => {
+      taskItem.classList.add('task-editing');
+      taskMenu.classList.add('hidden');
+
+      if (taskItem.classList.contains('task-editing')) {
+        trashIcon.classList.remove('hidden');
+        trashIcon.classList.add('show');
+
+        taskDescription.contentEditable = true;
+        taskDescription.focus();
+
+        trashIcon.addEventListener('click', () => {
```

Comment:
- [ ] Deleting one item from the list should re-arrange the whole list and reassign new indexes starting from 1, kindly make sure to fix this as for now the indexes are not re-arranging

![Screenshot_20230516_062210](https://github.com/victormc13/To-Do-List/assets/25227022/be680b17-7aca-47cf-9b4e-4191f0dc6637)